x-pack/legacy/plugins/apm/public/components/app/ServiceDetails/ServiceMetrics.tsx

```diff
@@ -21,25 +21,17 @@ export function ServiceMetrics({ urlParams, agentName }: ServiceMetricsProps) {
   const { start, end } = urlParams;
   return (
     <React.Fragment>
-      <SyncChartGroup
-        render={hoverXHandlers => (
-          <EuiFlexGrid columns={2} gutterSize="s">
-            {data.charts.map(chart => (
-              <EuiFlexItem key={chart.key}>
-                <EuiPanel>
-                  <MetricsChart
-                    start={start}
-                    end={end}
-                    chart={chart}
-                    hoverXHandlers={hoverXHandlers}
-                  />
-                </EuiPanel>
-              </EuiFlexItem>
-            ))}
-          </EuiFlexGrid>
-        )}
-      />
-
+      <EuiFlexGrid columns={2} gutterSize="s">
+        {data.charts.map(chart => (
+          <EuiFlexItem key={chart.key}>
+            <EuiPanel>
+              <ChartsSyncContextProvider>
+                <MetricsChart start={start} end={end} chart={chart} />
+              </ChartsSyncContextProvider>
+            </EuiPanel>
+          </EuiFlexItem>
+        ))}
+      </EuiFlexGrid>
```

Comment:
Yay! No more HOCs!

Reply:
... or rather: render props. But same same. Both legacy now we have hooks and context :)